src/main/java/UI/ProfileReviewPanel.java

```diff
@@ -0,0 +1,159 @@
+        usernameReviewed = new JTextField(15);
+        usernameInfo = new JPanel();
+        usernameInfo.add(usernameLabel);
+        usernameInfo.add(usernameReviewed);
+
+        ratingChosen = new JLabel("");
+        successfulReview = new JLabel("");
+        errorMessage = new JLabel("");
+
+        JLabel ratingLabel = new JLabel("Give them a rating between 1 and 5: ");
+        rating1 = new JButton("1");
+        rating2 = new JButton("2");
+        rating3 = new JButton("3");
+        rating4 = new JButton("4");
+        rating5 = new JButton("5");
+        ratingInfo = new RatingPanel(ratingLabel, rating1, rating2, rating3, rating4, rating5);
+
+        rating1.addActionListener(this);
+        rating2.addActionListener(this);
+        rating3.addActionListener(this);
+        rating4.addActionListener(this);
+        rating5.addActionListener(this);
+
+        submit = new JButton("Submit review");
+        submit.addActionListener(this);
+
+        reset = new JButton("Reset");
+        reset.addActionListener(this);
+
+        this.add(usernameInfo);
+        this.add(ratingInfo);
+        this.add(ratingChosen);
+        this.add(submit);
+        this.add(reset);
+        this.add(successfulReview);
+        this.add(errorMessage);
+    }
+
+    /**
+     * This aligns each JLabel, JButton, and JPanel so that they are placed correctly within the ProfileReviewPanel.
+     */
+    private void setUpLayout() {
+        layout.putConstraint(SpringLayout.NORTH, usernameInfo, 0, SpringLayout.NORTH, this);
+        layout.putConstraint(SpringLayout.NORTH, ratingInfo, 10, SpringLayout.SOUTH, usernameInfo);
+        layout.putConstraint(SpringLayout.NORTH, ratingChosen, 0, SpringLayout.SOUTH, ratingInfo);
+        layout.putConstraint(SpringLayout.WEST, ratingChosen, 5, SpringLayout.WEST, ratingInfo);
+        layout.putConstraint(SpringLayout.NORTH, submit, 30, SpringLayout.SOUTH, ratingInfo);
+        layout.putConstraint(SpringLayout.NORTH, reset, 30, SpringLayout.SOUTH, ratingInfo);
+        layout.putConstraint(SpringLayout.WEST, reset, 20, SpringLayout.EAST, submit);
+        layout.putConstraint(SpringLayout.NORTH, successfulReview, 15, SpringLayout.SOUTH, submit);
+        layout.putConstraint(SpringLayout.WEST, successfulReview, 5, SpringLayout.WEST, ratingInfo);
+        layout.putConstraint(SpringLayout.NORTH, errorMessage, 15, SpringLayout.SOUTH, submit);
+        layout.putConstraint(SpringLayout.WEST, errorMessage, 5, SpringLayout.WEST, ratingInfo);
+    }
+
+    /**
+     * This updates the panel depending on what event occurred.
+     * @param e The event to be processed.
+     */
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        if (e.getSource() == rating1) {
+            rating = 1;
+            ratingChosen.setText("Rating set to 1.");
+            errorMessage.setText("");
+            successfulReview.setText("");
+        } else if (e.getSource() == rating2) {
```

Comment:
Since the buttons are each just a number for the given rating, it might be cleaner to create a ratingButton class, and have one if statement here that checks if the button clicked is a ratingButton and uses e.getSource().getText(), or an attribute of the ratingButton class for the rest of the operations because all 5 if-elses for each rating are the same.

Reply:
Also, you should look into why the checks are failing